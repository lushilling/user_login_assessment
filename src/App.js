import React from 'react';
import './App.css';
import axios from 'axios';
import { BrowserRouter as Router, Route } from "react-router-dom";

import NavigationBar from './Components/NavigationBar';
import Homepage from './Components/Homepage';
import Register from './Components/Register';

export default class App extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      data: []
    };
  }

  onLoad = () => {
    axios.get("http://localhost:5000/user/all")
      .then(response => {
        this.setState({
          data: response.data
        })
      })
  }

  componentDidMount() {
    this.onLoad();
  }


  render() {
    return (
      <div >
        <Router>

          <NavigationBar />

          <Route exact path="/" component={Homepage} />

          <Route path="/Register" component={Register} render={() => <Register onLoadFunction={this.onload} data={this.state.data} />} />

        </Router>
      </div >
    )
  }
}


// <Route exact path="/" render={() => <RecipeList onLoadFunction={this.onload} data={this.state.data} />} />

// <Route path="/AddRecipe" component={AddRecipe} render={() => <AddRecipe onLoadFunction={this.onload} data={this.state.data} />} />

// <Route path="/MoreInfo" component={MoreInfo} render={() => <MoreInfo onLoadFunction={this.onload} data={this.state.data} />} />

// {this.state.data.map((recipe) => (

//   <Route path={"/" + recipe.name} render={() => <MoreInfo passedFunction={this.onLoad}
//     name={recipe.name}
//     description={recipe.description}
//     ingredients={recipe.ingredients}
//     image={recipe.image} />} />

// ))}