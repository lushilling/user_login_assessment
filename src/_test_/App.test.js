import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Route } from "react-router-dom";

import App from '../App';
import NavigationBar from '../Components/NavigationBar';


it('renders App', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot();
})


it('renders router without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders navigation bar without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NavigationBar />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders app without crashing', () => {
    shallow(<App />);
});

it('renders routingExample comp', () => {
    const wrapper = shallow(<App />);
    const content = <NavigationBar />;
    expect(wrapper.contains(content)).toEqual(true);
});

//Testing rendering a Prop
it("render a prop", () => {
    const wrapper = shallow(<App username="A Name" />);
    expect(wrapper.instance().props.username).toEqual("A Name");
    expect(wrapper).toMatchSnapshot();
});