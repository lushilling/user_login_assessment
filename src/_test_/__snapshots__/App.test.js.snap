// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a prop 1`] = `ShallowWrapper {}`;

exports[`renders App 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        Assessment
      </a>
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="./Register"
              onClick={[Function]}
            >
              Register
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="./Login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="./UserList"
              onClick={[Function]}
            >
              User List
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div>
    <p />
    <br />
    <p>
      On this React app you can register a user, login as a user and view a list of users
    </p>
  </div>
</div>
`;
